cliquet/scripts/cliquet.py

```diff
@@ -15,10 +17,18 @@ def deprecated_init(env):
 
 def init_schema(env):
     registry = env['registry']
+    settings = registry.settings
+    readonly_backends = ('storage', 'permission')
+    readonly_mode = asbool(settings.get('readonly', False))
 
     for backend in ('cache', 'storage', 'permission'):
         if hasattr(registry, backend):
-            getattr(registry, backend).initialize_schema()
+            if readonly_mode and backend in readonly_backends:
+                message = ('Cannot migrate the %s backend while '
+                           'in readonly mode.' % backend)
+                warnings.warn(message, Warning)
```

Comment:
Don't pass the base Warning class here, warn() uses a UserWarning by default. If you want a specific Warning class you need to pass a XXXWarning or create your own